OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -31,14 +31,10 @@ public class WithInfantryBodyInfo : UpgradableTraitInfo, IRenderActorPreviewSpri
 
 		public IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer init, RenderSpritesInfo rs, string image, int facings, PaletteReference p)
 		{
-			var facing = 0;
 			var ifacing = init.Actor.TraitInfoOrDefault<IFacingInfo>();
-			if (ifacing != null)
-				facing = init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : ifacing.GetInitialFacing();
-
-			var anim = new Animation(init.World, image, () => facing);
+			var anim = new Animation(init.World, image, () => ifacing != null ? init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : ifacing.GetInitialFacing() : 0);
```

Comment:
Actually this comments apply to most of the uses of `init.Contains<FacingInit>()`.  Hiding the dynamic-ness of the facing inside `DynamicFacingInit.Value` makes it easy to mess up the use of facings.  I think it would be much better to pull out a helper method that takes an init dict, a fallback facing (from InitialFacing or elsewhere), and returns a `Func<int>`.  Internally it would explicitly check for `DynamicFacingInit` vs `FacingInit`.